src/components/OnboardingComponents/SemanticSearch/ArticleDialog.tsx

```diff
@@ -0,0 +1,210 @@
+  const [pageNumber, setPageNumber] = useState(1);
+  const [containerWidth, setContainerWidth] = useState(0);
+  const [zoom, setZoom] = useState(1);
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const observer = new ResizeObserver((entries: ResizeObserverEntry[]) => {
+      const entry = entries[0];
+      if (entry) {
+        setContainerWidth(entry.contentRect.width);
+      }
+    });
+
+    if (containerRef.current) {
+      observer.observe(containerRef.current);
+    }
+
+    return () => {
+      observer.disconnect();
+    };
+  }, []);
+
+  const goToNextPage = useCallback(() => {
+    setPageNumber((prevPageNumber) => Math.min(prevPageNumber + 1, numPages));
+  }, [numPages]);
+
+  const goToPreviousPage = useCallback(() => {
+    setPageNumber((prevPageNumber) => Math.max(prevPageNumber - 1, 1));
+  }, []);
+
+  const handleZoomIn = useCallback(() => {
+    setZoom((prevZoom) => Math.min(prevZoom + 0.25, 3));
+  }, []);
+
+  const handleZoomOut = useCallback(() => {
+    setZoom((prevZoom) => Math.max(prevZoom - 0.25, 0.25));
+  }, []);
+
+  const handleZoomReset = useCallback(() => {
+    setZoom(1);
+  }, []);
+
+  // Keyboard shortcuts for zoom and navigation
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (!open) return;
+
+      switch (event.key) {
+        case '+':
+        case '=':
+          if (event.ctrlKey || event.metaKey) {
+            event.preventDefault();
+            handleZoomIn();
+          }
+          break;
+        case '-':
+          if (event.ctrlKey || event.metaKey) {
+            event.preventDefault();
+            handleZoomOut();
+          }
+          break;
+        case '0':
+          if (event.ctrlKey || event.metaKey) {
+            event.preventDefault();
+            handleZoomReset();
+          }
+          break;
+        case 'ArrowLeft':
+          if (event.ctrlKey || event.metaKey) {
+            event.preventDefault();
+            goToPreviousPage();
+          }
+          break;
+        case 'ArrowRight':
+          if (event.ctrlKey || event.metaKey) {
+            event.preventDefault();
+            goToNextPage();
+          }
+          break;
+      }
+    };
+
+    document.addEventListener('keydown', handleKeyDown);
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [open, handleZoomIn, handleZoomOut, handleZoomReset, goToPreviousPage, goToNextPage]);
+
+  if (!article) {
+    return <></>;
+  }
+
+  function onDocumentLoadSuccess({numPages: nextNumPages}: {numPages: number}) {
+    setNumPages(nextNumPages);
+    setPageNumber(1); // Reset to first page on new document load
+  }
+
+  return (
+    <Dialog
+      open={open}
+      onClose={onClose}
+      maxWidth='lg'
+      fullWidth
+      scroll='paper'
+      sx={{'& .MuiDialog-paper': {height: '90vh'}}}
+    >
+      <DialogTitle sx={{m: 0, p: 6, pb: 5}}>
```

Comment:
Please use to full keywords so the style is easier to read and understand (even for non-expert developers).
Also expand the sx prop so it is easier to read.

```suggestion
      <DialogTitle
        sx={{
          margin: 0,
          padding: 6,
          paddingBottom: 5
        }}
      >
```

Reply:
Solved

>